src/TodoList.js

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import TodoItems from './TodoItems'
+import Footer from "./Footer";
+
+class TodoList extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            value: '',
+            items: [],
+            filter: 'all'
+        };
+
+
+    }
+
+    onChange = (event) => {
+        this.setState({value:event.target.value})
+    };
+
+
+    onSubmit = (event) => {
+        event.preventDefault();
+
+        let newItem = {
+            text: this.state.value,
+            key: Date.now(),
+            done: false,
+            visibility: null
+        };
+
+        this.setState((prevState) => {
+            return {
+                value: '',
+                items:prevState.items.concat(newItem)}
+        });
+    };
+
+    remove(index) {
+        const items = this.state.items.filter((item, i) => i !== index);
```

Comment:
It would be great to pass `id` instead of `index`. Working with indices may bring some issues with dynamic data.
```suggestion
        const items = this.state.items.filter((item) => item.id !== id);
```